fact/src/main.rs

```diff
@@ -1,9 +1,28 @@
+use std::io::Write;
+use std::str::FromStr;
+
 use clap::Parser;
-use env_logger::Env;
+use log::LevelFilter;
 
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
-    env_logger::Builder::from_env(Env::default().filter_or("FACT_LOGLEVEL", "info")).init();
+    let log_level = std::env::var("FACT_LOGLEVEL").unwrap_or("info".to_owned());
+    let log_level = LevelFilter::from_str(&log_level)?;
+    env_logger::Builder::new()
+        .filter_level(log_level)
+        .format(move |buf, record| {
+            write!(buf, "[{:<5} {}] ", record.level(), buf.timestamp_seconds())?;
+            if matches!(log_level, LevelFilter::Debug | LevelFilter::Trace) {
```

Comment:
neat! not seen this `matches!` macro before

Reply:
`cargo clippy` is a hell of a drug maaaaaan.